jupyterlab_git/git.py

```diff
@@ -1112,6 +1112,48 @@ def remote_add(self, top_repo_path, url, name=DEFAULT_REMOTE_NAME):
                 "message": my_error.decode("utf-8").strip()
             }
 
+    async def ensure_gitignore(self, top_repo_path):
+        """Handle call to ensure .gitignore file exists and the 
+        next append will be on a new line (this means an empty file
+        or a file ending with \n).
+
+        top_repo_path: str
+            Top Git repository path
+        """
+        try:            
+            gitignore = os.path.join(top_repo_path, ".gitignore")
+            if not os.path.exists(gitignore):
+                open(gitignore, "w").close()
+            elif os.path.getsize(gitignore) != 0:
+                f = open(gitignore, "r")
+                if (f.read()[-1] != "\n"):
+                    f = open(gitignore, "a")
+                    f.write('\n')
+                f.close()
+            return {"code": 0}
+        except:
+            return {"code": -1}
```

Comment:
```suggestion
        try:            
            gitignore = pathlib.Path(top_repo_path) / ".gitignore"
            if not gitignore.exists():
                gitignore.touch()
            elif gitignore.stat().st_size > 0:
                content = gitignore.read_text()
                if (content[-1] != "\n"):
                    with gitignore.open("a") as f:
                        f.write('\n')
        except BaseException as error:
            return {"code": -1, "message": str(error)}
        else:
            return {"code": 0}
```

I'm a big supporter of [pathlib.Path](https://docs.python.org/3/library/pathlib.html). It has nice feature like `touch`.

You should use context opening as much as possible. This ensures a proper file closure (the original code as some unbalanced open/close).

Reply:
Thx for the review. Last commit implements this.